### PR TITLE
NAS-133042 / 25.04 / Lock out plugins when in GPOS STIG mode

### PR DESCRIPTION
**Pull request by anodos325**
Certain TrueNAS features are incompatible with requirements for
the GPOS STIG. This commit makes changes to our RBAC framework
to decrease effective privileges granted to all credentials to
prevent configuring non-compliant services and to give UI hints
about which features to disable.

As a result of these changes, the allowlist column is dropped
from the privilege table. Internally our credentials still
generate an allowlist in order to perform RBAC checks

### CODE REVIEW

**Review by billohanlon — APPROVED** [review]
LGTM.

**Review by yocalebo — APPROVED** [review]
Let's get it merged sooner rather later.

File: tests/api2/test_stig.py [hunk 1]
```diff
@@ -2,11 +2,13 @@
 import pytest
 
 from middlewared.service_exception import CallError
-from middlewared.test.integration.assets.product import enable_stig, product_type, set_fips_available
+from middlewared.service_exception import ValidationErrors as Verr
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:5]: Trailing whitespace

File: tests/api2/test_stig.py [hunk 2]
```diff
@@ -2,11 +2,13 @@
 import pytest
 
 from middlewared.service_exception import CallError
-from middlewared.test.integration.assets.product import enable_stig, product_type, set_fips_available
+from middlewared.service_exception import ValidationErrors as Verr 
+from middlewared.test.integration.assets.account import user
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:6]: `user` is unused

File: tests/api2/test_stig.py [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
+from middlewared.test.integration.assets.product import product_type, set_fips_available
 from middlewared.test.integration.assets.two_factor_auth import (
     enabled_twofactor_auth, get_user_secret, get_2fa_totp_token
 )
-from middlewared.test.integration.utils import call, client
+from middlewared.test.integration.utils import call, client, ssh
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:11]: `ssh` is unused and make sure to get the trailing whitespace

File: tests/api2/test_stig.py [hunk 4]
```diff
@@ -76,6 +62,56 @@ def two_factor_full_admin(two_factor_enabled, unprivileged_user_fixture):
...
+                    yield {
+                        'connection': c,
+                        'user_obj': user_obj,
+                        'secret': secret
+                    }
+                finally:
+                    c.call('system.security.update', {'enable_fips': False, 'enable_gpos_stig': False}, job=True)
+
+
+# The order of the following tests is significant. We gradully fixtures that have module scope
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:112]: If you wish yous can 'gradully' fix the typos.
Enough cheeky business.  Maybe:
`The order of the following tests are significant. We gradually add fixtures that have module scope`

I do love comments!

File: tests/api2/test_stig.py [hunk 5]
```diff
@@ -107,28 +143,58 @@ def test_no_current_cred_no_2fa(enterprise_product, two_factor_full_admin):
...
+    # We should also be able to create a new websocket connection
+    # The previous one was created before enabling STIG
+    with client(auth=None) as c:
+        do_stig_auth(c, setup_stig['user_obj'], setup_stig['secret'])
+        c.call('system.info')
+
+
+def test_stig_roles_decrease(setup_stig):
+
+    # We need new webosocket connection to verify that privileges
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:169]: typo: `webosocket`

File: tests/api2/test_stig.py [hunk 6]
```diff
@@ -107,28 +143,58 @@ def test_no_current_cred_no_2fa(enterprise_product, two_factor_full_admin):
...
+                assert role['name'] in me['privilege']['roles']
+            else:
+                assert role['name'] not in me['privilege']['roles']
+
+        assert me['privilege']['web_shell'] is False
+        assert me['privilege']['webui_access'] is True
+
+
+def test_stig_smb_auth_disabled(setup_stig):
+    # We need new webosocket connection to verify that privileges
```
Comment (mgrimesix) [inline tests/api2/test_stig.py:186]: cut-n-pasted the typo

File: tests/unit/test_role_manager.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        assert role_to_check.full_admin is False
+
+
+@pytest.mark.parametrize('role,method,enabled_stig_type,resources', [
+    ('READONLY_ADMIN', 'CALL', None, READONLY_ADMIN_METHODS),
+    ('READONLY_ADMIN', 'CALL', security.STIGType.GPOS, READONLY_ADMIN_METHODS),
+    ('FULL_ADMIN', '*', None, {'*'}),
+    ('FULL_ADMIN', 'CALL', security.STIGType.GPOS, FULL_ADMIN_STIG),
+])
+def test__roles_have_corect_allowlist(role_manager, role, method, enabled_stig_type, resources):
```
Comment (mgrimesix) [inline tests/unit/test_role_manager.py:66]: Might as well fix the typo in the function name.

**Review by mgrimesix — APPROVED** [review]
Only flake8 and typos.  Nice!
You may clean up 'gradully'.  😉

**Pull request merged**